TESTS/mbed_hal/qspi/qspi_test_utils.cpp

```diff
@@ -0,0 +1,233 @@
+    qspi_status_t ret;
+    uint32_t curr_time;
+
+    const ticker_data_t *const ticker = get_us_ticker_data();
+    const uint32_t start = ticker_read(ticker);
+
+    memset(reg, 255, QSPI_STATUS_REG_SIZE);
+    do {
+        ret = read_register(STATUS_REG, reg, QSPI_STATUS_REG_SIZE, qspi);
+        curr_time = ticker_read(ticker);
+    } while (((reg[0] & STATUS_BIT_WIP) != 0) && ((curr_time - start) < time_us));
+
+    if (((reg[0] & STATUS_BIT_WIP) == 0) && (ret == QSPI_STATUS_OK)) {
+        return sOK;
+    } else if (ret != QSPI_STATUS_OK) {
+        return sError;
+    } else if ((curr_time - start) >= time_us) {
+        return sTimeout;
+    }
+    return sUnknown;
+}
+
+void flash_init(Qspi &qspi)
+{
+    uint8_t status[QSPI_STATUS_REG_SIZE];
+    qspi_status_t ret;
+
+    qspi.cmd.build(QSPI_CMD_RDSR);
+    ret = qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, status, QSPI_STATUS_REG_SIZE);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    qspi.cmd.build(QSPI_CMD_RSTEN);
+    ret = qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    qspi.cmd.build(QSPI_CMD_RST);
+    ret = qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    WAIT_FOR(WAIT_MAX_TIME, qspi);
+}
+
+
+qspi_status_t write_enable(Qspi &qspi)
+{
+    uint8_t reg[QSPI_STATUS_REG_SIZE];
+    qspi.cmd.build(QSPI_CMD_WREN);
+
+    if (qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    memset(reg, 0, QSPI_STATUS_REG_SIZE);
+    if (read_register(STATUS_REG, reg, QSPI_STATUS_REG_SIZE, qspi) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+
+    return ((reg[0] & STATUS_BIT_WEL) != 0 ? QSPI_STATUS_OK : QSPI_STATUS_ERROR);
+}
+
+qspi_status_t write_disable(Qspi &qspi)
+{
+    uint8_t reg[QSPI_STATUS_REG_SIZE];
+    qspi.cmd.build(QSPI_CMD_WRDI);
+
+    if (qspi_command_transfer(&qspi.handle, qspi.cmd.get(), NULL, 0, NULL, 0) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+    WAIT_FOR(WRSR_MAX_TIME, qspi);
+
+    memset(reg, 0, QSPI_STATUS_REG_SIZE);
+    if (read_register(STATUS_REG, reg, QSPI_STATUS_REG_SIZE, qspi) != QSPI_STATUS_OK) {
+        return QSPI_STATUS_ERROR;
+    }
+
+    return ((reg[0] & STATUS_BIT_WEL) == 0 ? QSPI_STATUS_OK : QSPI_STATUS_ERROR);
+}
+
+void log_register(uint32_t cmd, uint32_t reg_size, Qspi &qspi)
+{
+    qspi_status_t ret;
+    static uint8_t reg[QSPI_MAX_REG_SIZE];
+
+    ret = read_register(cmd, reg, reg_size, qspi);
+    TEST_ASSERT_EQUAL(QSPI_STATUS_OK, ret);
+
+    for (uint32_t j = 0; j < reg_size; j++) {
+        utest_printf("register byte %u data: ", j);
+        for(int i = 0; i < 8; i++) {
+            utest_printf("%s ", ((reg[j] & (1 << i)) & 0xFF) == 0 ? "0" : "1");
```

Comment:
Doesn't it make more sense to print the register contents MSB first? I.e. do a `(1 << (7-i))`   instead.